src/RoomsModel.cpp

```diff
@@ -0,0 +1,57 @@
+#include "RoomsModel.h"
+
+#include "Cache_p.h"
+#include "CompletionModelRoles.h"
+
+RoomsModel::RoomsModel(QObject *parent)
+  : QAbstractListModel(parent)
+{
+        rooms_    = cache::joinedRooms();
+        roomInfos = cache::getRoomInfo(rooms_);
+
+        for (const auto &r : rooms_) {
+                auto roomAliasesList = cache::client()->getRoomAliases(r);
+
+                if (roomAliasesList) {
+                        roomAliases.push_back(QString::fromStdString(roomAliasesList->alias));
+                        roomids.push_back(QString::fromStdString(r));
+                }
+        }
+}
+
+QHash<int, QByteArray>
+RoomsModel::roleNames() const
+{
+        return {{CompletionModel::CompletionRole, "completionRole"},
+                {CompletionModel::SearchRole, "searchRole"},
+                {CompletionModel::SearchRole2, "searchRole2"},
+                {Roles::RoomAlias, "roomAlias"},
+                {Roles::AvatarUrl, "avatarUrl"},
+                {Roles::RoomID, "roomid"},
+                {Roles::RoomName, "roomName"}};
+}
+
+QVariant
+RoomsModel::data(const QModelIndex &index, int role) const
+{
+        if (hasIndex(index.row(), index.column(), index.parent())) {
+                switch (role) {
+                case CompletionModel::CompletionRole:
+                        return QString("%1").arg(roomAliases[index.row()]);
```

Comment:
It would be nice, if you could return a markdown formatted matrix.to link here, so that it gets formatted as a link. You can copy that from the user model completer.